src/browser/browser.js

```diff
@@ -297,15 +299,27 @@ define((require, exports, module) => {
       key: 'appUpdateBanner',
       className: ClassSet({
         appupdatebanner: true,
-        active: immutableState.get('appUpdateAvailable')
+        active: immutableState.get('appUpdateAvailable') || immutableState.get('runtimeUpdateAvailable')
       }),
     }, [
       'Hey! An update just for you!',
       DOM.div({
         key: 'appUpdateButton',
         className: 'appupdatebutton',
-        onClick: e => sendEventToChrome('clear-cache-and-reload')
-      }, 'Apply')
+        onClick: e => {
+          if (immutableState.get('runtimeUpdateAvailable') && immutableState.get('runtimeUpdateAvailable')) {
+            // FIXME: Not supported yet
+            sendEventToChrome('clear-cache-and-restart')
+          }
+          if (immutableState.get('runtimeUpdateAvailable') && !immutableState.get('runtimeUpdateAvailable')) {
+            // FIXME: Not supported yet
+            sendEventToChrome('restart')
+          }
+          if (!immutableState.get('runtimeUpdateAvailable') && immutableState.get('runtimeUpdateAvailable')) {
+            sendEventToChrome('clear-cache-and-reload')
+          }
+        }
+      }, 'Apply' + (immutableState.get('runtimeUpdateAvailable') ? ' (restart required)' : ''))
```

Comment:
Note that some updates may depend on runtime updates so that would need to be expressed some how as well.